### PR TITLE
feat(link-modules, modules-sdk, types): link modules extra fields

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**

### CODE REVIEW

File: packages/types/src/modules-sdk/index.ts [hunk 1]
```diff
@@ -114,7 +114,40 @@ export type ModuleJoinerConfig = Omit<
   isLink?: boolean // If the module is a link module
   linkableKeys?: string[] // Keys that can be used to link to other modules
   isReadOnlyLink?: boolean // If true it expands a RemoteQuery property but doesn't have a pivot table
-  tableName?: string // Name of the pivot table. If not provided it is auto generated
+  databaseConfig?: {
+    tableName?: string // Name of the pivot table. If not provided it is auto generated
+    idPrefix?: string // Prefix for the id column. If not provided it is "link"
+    schema?: Record<
```
Comment (olivermrbl) [inline packages/types/src/modules-sdk/index.ts]: question: Is the type specific to MikroORM? Was specifically thinking about the `defaultRaw`

Should this be an ORM-agnostic config type?

Comment (olivermrbl) [reply packages/types/src/modules-sdk/index.ts]: Discard this comment. I realise the linking will be specific to MikroORM.

Comment (olivermrbl) [inline packages/types/src/modules-sdk/index.ts]: suggestion(non-blocking): `fields` or `extraFields`?

File: integration-tests/plugins/__tests__/link-modules/index.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  let links
+  const linkDefinition: ModuleJoinerConfig[] = [
+    {
+      serviceName: "linkServiceName",
+      isLink: true,
+      databaseConfig: {
+        tableName: "linkTableName",
+        idPrefix: "prefix",
+        schema: {
+          extra_field: {
```
Comment (olivermrbl) [inline integration-tests/plugins/__tests__/link-modules/index.ts:22]: praise: super nice!

File: packages/types/src/modules-sdk/index.ts [hunk 3]
```diff
@@ -114,7 +114,40 @@ export type ModuleJoinerConfig = Omit<
...
+          | "tinyint"
+          | "mediumint"
+          | "float"
+          | "double"
+          | "boolean"
+          | "decimal"
+          | "string"
+          | "uuid"
+          | "text"
+        defaultRaw?: string
```
Comment (olivermrbl) [inline packages/types/src/modules-sdk/index.ts]: suggestion(non-blocking): `defaultValue`

**Review by olivermrbl — APPROVED** [review]
LGTM!

File: packages/types/src/modules-sdk/index.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply packages/types/src/modules-sdk/index.ts]: will change it to `extraFields`

File: packages/types/src/modules-sdk/index.ts (hunk 3 above)
Comment (carlos-r-l-rodrigues) [reply packages/types/src/modules-sdk/index.ts]: `defaultValue` looks better

**Pull request merged**